File: 백준/단계 별로 풀어보기/1차원 배열/3052_나머지.py
import sys
input = sys.stdin.readline
IP = lambda: map(int, sys.stdin.readline().split())

def solve():
    remain = [False] * 42
    for i in range(10): remain[int(input()) % 42] = True
    print(sum(remain))
    
if "__main__" == __name__: solve()

File: baekjoon/단계 별로 풀어보기/1차원 배열/3052_나머지.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

int main()
{
	int remain[42]{0,};
	int command;
	int sum = 0;
	
	for (int i = 0; i < 10; i++)
	{
	    scanf("%d", &command);
	    remain[command % 42] = 1;
	}
	for (int i = 0; i < 42; i++) sum += remain[i];
	printf("%d", sum);
}